src/modules/pages/ViewPage/ApiItem.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react'
+import R from 'ramda'
+import PropTypes from 'prop-types'
+
+import transformFormValues from '../../../helpers/transformFormValues'
+import getUpdatedEndpoint from '../../../helpers/getUpdatedEndpoint'
+import isAnyEmpty from '../../../helpers/isAnyEmpty'
+
+import EndpointForm from '../../forms/EndpointForm/EndpointForm'
+import Preloader from '../../../components/Preloader/Preloader'
+
+const propTypes = {
+  api: PropTypes.object.isRequired,
+  deleteEndpoint: PropTypes.func.isRequired,
+  excludePlugin: PropTypes.func.isRequired,
+  fetchEndpointSchema: PropTypes.func.isRequired,
+  fillSelected: PropTypes.func.isRequired,
+  isAdmin: PropTypes.bool,
+  selectPlugin: PropTypes.func.isRequired,
+  fetchEndpoint: PropTypes.func.isRequired,
+  refreshEndpoints: PropTypes.func.isRequired,
+  resetEndpoint: PropTypes.func.isRequired,
+  updateEndpoint: PropTypes.func.isRequired,
+  location: PropTypes.object.isRequired
+}
+
+class ApiItem extends PureComponent {
+  componentDidMount () {
+    const endpointName = this.props.location.pathname.split('/preview/')[1]
+
+    this.props.resetEndpoint()
+    this.props.fetchEndpointSchema()
+    this.props.fetchEndpoint(endpointName)
+  }
+
+    fillSelected = (arr) => {
+      const selectedPlugins = arr.map(item => item.name)
+
+      this.props.fillSelected(selectedPlugins)
+    }
+
+    submit = values => R.compose(
+      this.props.updateEndpoint,
+      getUpdatedEndpoint(values)
+    )(this.props.selectedPlugins);
+
+    handleDelete = () => {
+      const needRedirect = true
+
+      this.props.deleteEndpoint(this.props.api, needRedirect)
+    };
+
+    render () {
+      if (isAnyEmpty([
+        this.props.api,
+        this.props.apiSchema
+      ])) return <Preloader />
```

Comment:
Apart from `PreLoader` component, do we render any error message when an error occurs in the fetching?